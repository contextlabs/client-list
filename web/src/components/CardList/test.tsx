import { render, screen } from '@testing-library/react';

import { CardList } from '.';

describe('<CardList />', () => {
  const props = {
    results: [
      {
        name: 'CardList Name',
        title: 'CardList Title',
        avatar: 'https://gravatar.com/4405735f6f3129e0286d9d43e7b460d0',
        nationality: 'Global',
        quote: 'Hello'
      }
    ],
    onClickNavigate: Function
  };

  it('should render the CardList as default', () => {
    const { container } = render(<CardList {...props} />);
    expect(container.firstChild).toMatchSnapshot();

    expect(
      screen.getByRole('heading', { name: /CardList Title/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: /CardList Name/i })
    ).toBeInTheDocument();
  });
});
