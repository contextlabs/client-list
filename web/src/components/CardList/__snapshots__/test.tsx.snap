// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should render the CardList as default 1`] = `
<div
  aria-label="CardList"
>
  <div
    class="flex flex-col justify-between p-4 m-4 my-16 mx-10 bg-white rounded-md border-l-4 border-teal-500 border-solid shadow-md sm:flex-row sm:my-8"
  >
    <div
      class="flex flex-col sm:flex-row"
    >
      <div>
        <img
          class="inline-block rounded-full w-12 h-12"
          src="https://gravatar.com/4405735f6f3129e0286d9d43e7b460d0"
        />
      </div>
      <div
        class="flex flex-col justify-between ml-4"
      >
        <h3
          class="text-base font-bold text-teal-500"
        >
          CardList Title
        </h3>
        <h2
          class="my-2 text-xl font-bold sm:my-0"
        >
          CardList Name
        </h2>
      </div>
    </div>
  </div>
</div>
`;
